### PR TITLE
Add debug features

### PR DESCRIPTION
**Pull request by nulinspiratie**
Add debug utilities by wrapping classes such that every function call is logged.

Good to go

### CODE REVIEW

File: qcodes/utils/debug.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        # print(f" | {time.perf_counter() - t0:.3g} s", file=log)
+        # log.flush()
+
+        return returnval
+
+    return _methodWrapper
+
+
+def logclass(
+    cls, methodsAsFunctions=False, log=None, logMatch=".*", logNotMatch="asdfnomatch"
```
Comment (maij) [inline qcodes/utils/debug.py:76]: Things start getting a bit cryptic from this point. What is the purpose of logMatch and logNotMatch? Why is logNotMatch defaulting to a random string `asdf`?

Comment (nulinspiratie) [reply qcodes/utils/debug.py]: Better?

Comment (maij) [reply qcodes/utils/debug.py]: Yeah, it's much clearer. I did a quick google for unmatchable regex strings and it came back with /(?!)/

I just looked into a 'single regex' construction for doing positive and negative matching (i.e. the union of positive matches and NOT the negative matches) and while it's not horribly complicated I agree that this is just easier.

I have 2 suggestions:
- Replace `asdfnomatch` with `(?!)` and mention it in the docstring
- Perhaps allow log_match/log_not_match to be given as a list of strings for non-regexperts, and then combine them à la
```
log_match = ['banana', 'berry', 'apple']
log_match_re = '(banana)|(berry)|(apple)'
```

Comment (nulinspiratie) [reply qcodes/utils/debug.py]: Like this?

**Review by maij — APPROVED** [review]
Nice!

**Pull request merged**